perfkitbenchmarker/pkb.py

```diff
@@ -195,6 +195,51 @@
 events.initialization_complete.connect(traces.RegisterAll)
 
 
+def _CreateBenchmarkRunList():
+  """Create a list of configs and states for each benchmark run to be scheduled.
+
+  Returns:
+    list of (args, run_status_list) pairs. Contains one pair per benchmark run
```

Comment:
> In the case of staged runs, the status actually represents whether the desired stage was run and whether it was successful for each benchmark. So making it part of the BenchmarkSpec may not make sense, since it's a state that has more to do with the current PKB process, and not the overall benchmark run (provision-prepare-run-cleanup-teardown).

Yeah - that's why I mentioned having a Benchmark class may be more appropriate. It could keep track of things like the current run stage and the status for that stage and have methods related to running itself.

> The only thing missing from a partial is the collector argument, which currently is only created once and shared for all benchmarks. It would need to be passed to this function, which somehow seems unclean to me.

You could pass it to the partial later - it doesn't need to be bound right away.

> But it's a somewhat sizable change from the current behavior, so I'd prefer to think a bit more and then make a separate PR.

That's what I was figuring - SGTM